PrebidMobile/PrebidMobile-core/src/main/java/org/prebid/mobile/Util.java

```diff
@@ -597,4 +639,187 @@ public static String generateInstreamUriForGam(String adUnit, HashSet<AdSize> si
+        String isPrebid = (String) callMethodOnObject(object, "getText", "isPrebid");
+        if ("1".equals(isPrebid)) {
+            String cacheId = (String) callMethodOnObject(object, "getText", "hb_cache_id_local");
+            PrebidNativeAd ad = PrebidNativeAd.create(cacheId);
+            if (ad != null) {
+                listener.onPrebidNativeLoaded(ad);
+            } else {
+                listener.onPrebidNativeNotValid();
+            }
+        } else {
+            listener.onPrebidNativeNotFound();
+        }
+    }
+
+    private static void findNativeInMoPubNativeAd(@NonNull Object object, @NonNull PrebidNativeAdListener listener) {
+        Object baseNativeAd = callMethodOnObject(object, "getBaseNativeAd");
+        LogUtil.d("Prebid", "" + baseNativeAd);
+        Boolean isPrebid = (Boolean) callMethodOnObject(baseNativeAd, "getExtra", "isPrebid");
+        if (isPrebid != null && isPrebid) {
+            String cacheId = (String) callMethodOnObject(baseNativeAd, "getExtra", "hb_cache_id_local");
+            if (CacheManager.isValid(cacheId)) {
+                PrebidNativeAd ad = PrebidNativeAd.create(cacheId);
+                if (ad != null) {
+                    listener.onPrebidNativeLoaded(ad);
+                    return;
+                }
+            } else {
+                listener.onPrebidNativeNotValid();
+            }
+        } else {
+            listener.onPrebidNativeNotFound();
+        }
+    }
+
+    private static void findMoPubNativeAd(@NonNull Object object, @NonNull PrebidNativeAdListener listener) {
+        Object baseNativeAd = callMethodOnObject(object, "getBaseNativeAd");
+        LogUtil.d("Prebid", "" + baseNativeAd);
+        Boolean isPrebid = (Boolean) callMethodOnObject(baseNativeAd, "getExtra", "isPrebid");
+        if (isPrebid != null && isPrebid) {
+            String cacheId = (String) callMethodOnObject(baseNativeAd, "getExtra", "hb_cache_id");
+            if (CacheManager.isValid(cacheId)) {
+                PrebidNativeAd ad = PrebidNativeAd.create(cacheId);
+                if (ad != null) {
+                    listener.onPrebidNativeLoaded(ad);
+                    return;
+                }
+            } else {
+                listener.onPrebidNativeNotValid();
+            }
+        } else {
+            listener.onPrebidNativeNotFound();
+        }
+    }
+
+
+    private static void findNativeInGAMPublisherAdView(@NonNull View adView, @NonNull PrebidNativeAdListener listener) {
+        List<WebView> webViewList = new ArrayList<>();
+        recursivelyFindWebViewList(adView, webViewList);
+        if (webViewList.size() == 0) {
+            listener.onPrebidNativeNotFound();
+        } else {
+            iterateWebViewListAsync(webViewList, webViewList.size() - 1, listener);
+        }
+
+    }
+
+    private static void iterateWebViewListAsync(final List<WebView> webViewList, final int index, final PrebidNativeAdListener listener) {
```

Comment:
I am not sure about this. It is already being copied at other places as well. If you want it to be moved in a single class (eg AdViewUtils, which I don't think it is, as this method serves the purpose of performing InAppNative related task), then kindly suggest where do you want it to be moved.

Reply:
I have created all these methods inside `AdViewUtils`. A truncated version was copied into Dr.Prebid because it is a different module.

I propose to make changes into `AdViewUtils` it should be a single point of working with HTML code

Reply:
Sure, will do that.

Reply:
@avohraa, is this issue fixed? I still see duplication

Reply:
@yoalex5 I have pushed the changes. Please check if that looks fine.